recipes/libffi/all/conanfile.py

```diff
@@ -20,171 +22,57 @@ class LibffiConan(ConanFile):
-
-        types_c_src = os.path.join(self._source_subfolder, "src", "types.c")
-        tools.replace_in_file(types_c_src,
-                              "#include <ffi_common.h>",
-                              "#include <ffi_common.h>\n"
-                              "\n"
-                              "#include <complex.h>")
-        tools.replace_in_file(types_c_src,
-                              "FFI_COMPLEX_TYPEDEF(name, type, maybe_const)",
-                              "FFI_COMPLEX_TYPEDEF(name, complex_type, maybe_const)")
-        tools.replace_in_file(types_c_src,
-                              "_Complex type",
-                              "complex_type")
-        tools.replace_in_file(types_c_src,
-                              "#ifdef FFI_TARGET_HAS_COMPLEX_TYPE",
-                              "#ifdef _MSC_VER"
-                              "\n#  define FLOAT_COMPLEX _C_float_complex"
-                              "\n#  define DOUBLE_COMPLEX _C_double_complex"
-                              "\n#  define LDOUBLE_COMPLEX _C_ldouble_complex"
-                              "\n#else"
-                              "\n#  define FLOAT_COMPLEX float _Complex"
-                              "\n#  define DOUBLE_COMPLEX double _Complex"
-                              "\n#  define LDOUBLE_COMPLEX long double _Complex"
-                              "\n#endif"
-                              "\n"
-                              "\n#ifdef FFI_TARGET_HAS_COMPLEX_TYPE")
-        tools.replace_in_file(types_c_src,
-                              "FFI_COMPLEX_TYPEDEF(float, float, const)",
-                              "FFI_COMPLEX_TYPEDEF(float, FLOAT_COMPLEX, const)")
-        tools.replace_in_file(types_c_src,
-                              "FFI_COMPLEX_TYPEDEF(double, double, const)",
-                              "FFI_COMPLEX_TYPEDEF(double, DOUBLE_COMPLEX, const)")
-        tools.replace_in_file(types_c_src,
-                              "FFI_COMPLEX_TYPEDEF(longdouble, long double, FFI_LDBL_CONST)",
-                              "FFI_COMPLEX_TYPEDEF(longdouble, LDOUBLE_COMPLEX, FFI_LDBL_CONST)")
         if self.settings.os == "Macos":
             tools.replace_in_file(configure_path, r"-install_name \$rpath/", "-install_name ")
 
-        if self.settings.compiler == "clang" and float(str(self.settings.compiler.version)) >= 7.0:
-            # https://android.googlesource.com/platform/external/libffi/+/ca22c3cb49a8cca299828c5ffad6fcfa76fdfa77
-            sysv_s_src = os.path.join(self._source_subfolder, "src", "arm", "sysv.S")
-            tools.replace_in_file(sysv_s_src, "fldmiad", "vldmia")
-            tools.replace_in_file(sysv_s_src, "fstmiad", "vstmia")
-            tools.replace_in_file(sysv_s_src, "fstmfdd\tsp!,", "vpush")
+        if Version(self.version) < "3.3":
+            if self.settings.compiler == "clang" and Version(str(self.settings.compiler.version)) >= 7.0:
```

Comment:
maybe extract to patch as well?

Reply:
@madebr do you want to make this change or is this ready to be merged?

Reply:
I'll do the change, but is it ok if the patch is only used in the recipe and not in the `conandata.yml`?

Reply:
I think it better to have all patches in conadata.yml. If that makes the recipe more complex, better keep it as is. Thanks!

Reply:
Okay, ready to merge then!